### DISCUSSION TITLE
Extracting estimated SNP effects

### DISCUSSION THREAD

**Discussion by david-cros** [tpook92/MoBPS#6] [Q&A]
Hello,

I use rrblup to make genomic predictions in `breeding.diploid` and I want to use the estimated marker effects, but I didn't find how to access them. Could you tell me please ?

I make the genomic predictions with:
`FGS = breeding.diploid(FGS, bve=T, rrblup.bve=TRUE, bve.cohorts = "SSCTrandom1", bve.insert.cohorts=c("Fn_with_SSCT_phenotypes_For_GS1"), remove.effect.position=TRUE, store.bve.data=T,                                          add.gen=2, estimate.u=TRUE)`
(I also used `fast.uhat=F` but I don't find anything looking like estimated marker effects in FGS).

Thanks

David

**Comment by tpook92** [comment]
Dear David, 
sorry for the late response as I was on holiday last week. 

rrBLUP.bve does use the rrBLUP R-Paket. However, on default this will use a regular GBLUP. To calculate a ridge regression BLUP you have to set the parameter estimate.u in breeding.diploid() to TRUE. This functionality is available for the direct BVE model (which is the BVE default) and rrBLUP.bve. Estimated SNP effects will be stored in population$info$u_hat / FGS$info$u_hat respectively. This element contains a list with each list element being one ridge regression estimated and a new list element added for each estimation run.

In the current live version (1.8.09) this actually does not work when using less than all markers (e.g. a specific array or remove.effect.position to TRUE). You can find a development version of the package with all functionality at https://github.com/tpook92/MoBPS/tree/master/development. I am currently working on a major update regarding pen / litter effects which is not 100% ready - that's why i do not want to upload this as the new version but only in the development branch for now. All functionality from 1.8.09 is working in this development version.

Best regards
Torsten
